### PR TITLE
updating test actions to use promises

### PR DESCRIPTION
**Pull request by jasonpet**
handles part of issue #932 
the others are handled in PR https://github.com/openwhisk/openwhisk-catalog/pull/138

### CODE REVIEW

**Comment by jasonpet** [comment]
pg2#153

File: tests/dat/actions/wcbin.js [hunk 1]
```diff
@@ -5,25 +5,26 @@
...
-    whisk.invoke({
-        name : 'wc',
-        parameters : {
-            payload : str
-        },
-        blocking : true,
-        next : function(error, activation) {
+
+    var promise = new Promise(function(resolve, reject) {
+        whisk.invoke({
```
Comment (bjustin-ibm) [inline tests/dat/actions/wcbin.js]: @jasonpet What do you think about this instead? I _believe_ it is functionally the same (minus one `console.log()` in the error condition) and does not require you to explicitly instantiate a promise:

```javascript
function main(params) {
    var str = params.payload;
    console.log("The payload is '" + str + "'");

    return whisk.invoke({
            name: 'wc',
            parameters: { payload: str },
            blocking: true})
      .then(function (activation) {
            console.log('activation:', activation);
            var wordsInDecimal = activation.result.count;
            var wordsInBinary = wordsInDecimal.toString(2) + ' (base 2)';

            return { binaryCount: wordsInBinary };
        });
    });
}
```

Comment (jasonpet) [inline tests/dat/actions/wcbin.js]: I like it and actually started with this exact same syntax:)  Started second guessing myself though and changed it.  Will test it out and go with this if it works.

**Comment by bjustin-ibm** [comment]
Rebased/merged as 8cceedec5e19b7d61a6291d1c7cc4efa2a0994ce

**Pull request closed without merging**